### PR TITLE
Refactor Patient mode for extended name results, V1 maintained AB#15020

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes or Implements [AB#15020](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15020)

## Description

1. Update the V2 patient model to include new properties for Legal, Common and Preferred names.
2. The V1 endpoint should operate as is and no new functionality should be added to the code base.
3. Unit tests are updated to accommodate the new structures.

## Testing

- [X] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes
PatientModel to DependentInformation modeling
image

User's PatientModel
image

New PatientModel (V2):
image

Old PatientModel (V1):
image

Swagger Link:
Local: http://localhost:3002/swagger/index.html?urls.primaryName=V1

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Common/src/Delegates/ClientRegistriesDelegate.cs [hunk 1]
```diff
@@ -378,26 +378,15 @@ private bool PopulateNames(HCIM_IN_GetDemographicsResponseIdentifiedPerson retri
                 }
             }
 
-            PN nameSection = (documentedName ?? legalName)!;
```
Comment (BrianMaki) [inline Apps/Common/src/Delegates/ClientRegistriesDelegate.cs:381]: Isn't ClientRegistriesDelegete in common pointing to v1?  Isn't v1 supposed to be left untouched?

File: Apps/Patient/src/Controllers/PatientController.cs [hunk 2]
```diff
@@ -69,9 +75,14 @@ public PatientController(IPatientService patientService, Services.IPatientServic
         [ApiVersion("1.0")]
         [Route("{hdid}")]
         [Authorize(Policy = PatientPolicy.Read)]
-        public async Task<RequestResult<PatientModel>> GetPatient(string hdid)
+        public async Task<RequestResult<PatientModelV1>> GetPatient(string hdid)
```
Comment (BrianMaki) [inline Apps/Patient/src/Controllers/PatientController.cs:78]: Could we have not left this untouched and changed GetPatientV2 to return PatientModelV2 instead?

Note: eventually once we have problem details exception handling resolved, the ClientRegistriesDelegete and PatientService in Patient were going to be removed and pointed at Common.

File: Apps/Patient/src/Models/PatientModelV1.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+namespace HealthGateway.Patient.Models
+{
+    using System;
+    using System.Text.Json.Serialization;
+    using HealthGateway.Common.Models;
+
+    /// <summary>
+    /// V1 Model for patient data, required for V1 GetPatient Endpoint.
+    /// </summary>
+    public class PatientModelV1
```
Comment (BrianMaki) [inline Apps/Patient/src/Models/PatientModelV1.cs:25]: Wouldn't it better to name this to PatientModelV2 and have it pointed to GetPatientV2 instead?  Then you can leave GetPatient untoched.

Note: eventually once we have problem details exception handling resolved, the ClientRegistriesDelegete and PatientService in Patient are going to be removed and pointed at Common at which time, these changes would be applied there at that time.

**Review by BrianMaki — COMMENTED** [review]
I left some comments. You and others may disagree.

File: Apps/AdminWebClient/src/ClientApp/src/components/covidTreatmentAssessment/CovidTreatmentAssessment.vue [hunk 4]
```diff
@@ -151,7 +151,7 @@ export default class CovidTreatmentAssessmentComponent extends Vue {
     }
 
     private get patientFullName(): string {
-        return `${this.patient.firstname} ${this.patient.lastname} `;
+        return `${this.patient.preferredName.givenName} ${this.patient.preferredName.surname} `;
```
Comment (BrianMaki) [inline Apps/AdminWebClient/src/ClientApp/src/components/covidTreatmentAssessment/CovidTreatmentAssessment.vue:154]: This is AdminWebClient and it calls v2 so I don't think there should be any changes to this code.

File: Apps/AdminWebClient/src/ClientApp/src/models/patientData.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import Address from "@/models/address";
 import { StringISODate } from "@/models/dateWrapper";
+import Name from "@/models/Name";
 
 export default interface PatientData {
     hdid: string;
     personalhealthnumber: string;
-    firstname: string;
```
Comment (BrianMaki) [inline Apps/AdminWebClient/src/ClientApp/src/models/patientData.ts:7]: This is AdminWebClient and it calls v2 so I don't think there should be any changes to this code.

File: Apps/AdminWebClient/src/ClientApp/src/views/CovidCard.vue [hunk 6]
```diff
@@ -104,7 +104,7 @@ export default class CovidCardView extends Vue {
     }
 
     private get patientName(): string {
-        return `${this.searchResult?.patient?.firstname} ${this.searchResult?.patient?.lastname}`;
+        return `${this.searchResult?.patient?.preferredName.givenName} ${this.searchResult?.patient?.preferredName.surname}`;
```
Comment (BrianMaki) [inline Apps/AdminWebClient/src/ClientApp/src/views/CovidCard.vue:107]: This is AdminWebClient and it calls v2 so I don't think there should be any changes to this code.

File: Apps/AdminWebClient/src/ClientApp/src/models/patientData.ts [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
-    lastname: string;
+    preferredName: Name;
+    commonName: Name;
+    legalName: Name;
```
Comment (MikeLyttle) [inline Apps/AdminWebClient/src/ClientApp/src/models/patientData.ts:10]: The types should be `Name | undefined`.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

**Review by sslaws — CHANGES_REQUESTED** [review]

File: Apps/AdminWebClient/src/ClientApp/src/components/covidTreatmentAssessment/CovidTreatmentAssessment.vue [hunk 8]
```diff
@@ -477,7 +479,7 @@ export default class CovidTreatmentAssessmentComponent extends Vue {
                             </ValidationProvider>
                         </Card>
                         <Card
-                            title="7. Have you been diagnosed by a health care provider with a chronic condition 
+                            title="7. Have you been diagnosed by a health care provider with a chronic condition
```
[the remark this answers is not in the document]
Comment (BronzBierd) [reply Apps/AdminWebClient/src/ClientApp/src/components/covidTreatmentAssessment/CovidTreatmentAssessment.vue:482]: These are all automatic, I don't change them explicitly.

File: Apps/WebClient/src/ClientApp/src/models/patientData.ts [hunk 9]
```diff
@@ -3,13 +3,17 @@ import { StringISODate } from "@/models/dateWrapper";
 export default class PatientData {
     public hdid!: string;
     public personalhealthnumber!: string;
-    public firstname!: string;
-    public lastname!: string;
+    public preferredname!: Name;
+    public commonname: Name | undefined;
+    public legalname: Name | undefined;
```
[the remark this answers is not in the document]
Comment (BronzBierd) [reply Apps/WebClient/src/ClientApp/src/models/patientData.ts:8]: was following previous convention, I'll update

File: Apps/WebClient/src/ClientApp/src/models/patientData.ts [hunk 10]
```diff
@@ -3,13 +3,17 @@ import { StringISODate } from "@/models/dateWrapper";
...
     public birthdate?: StringISODate;
     public physicalAddress?: Address;
     public postalAddress?: Address;
 }
-
+export class Name {
+    public givenName!: string;
+    public surname!: string;
+}
 export class Address {
```
[the remark this answers is not in the document]
Comment (BronzBierd) [reply Apps/WebClient/src/ClientApp/src/models/patientData.ts:17]: Following previous structure of the file, but I'll take the opportunity to clean this up 😄

**Comment by BronzBierd** [comment]
Was closed due to a misunderstanding of the scope, the clarification that V2 can be performed without affecting V1 has been highlighted, a new cleaner PR will be created under branch `tasks/15020_2`

**Pull request closed without merging**